### DISCUSSION TITLE
Multi-database app with tests

### DISCUSSION THREAD

**Discussion by JohnnyWalkerDigital** [laravel/framework#39091] [Q&A] [answered]
I have a situation where I have an app that talks to one central database (which other apps also use) and has its own database for things relating only to itself. I want this app to be testable, locally and via something like Travis CI, which means I need all the migrations for both tables.

But when I deploy the app, I don't wish to destructively alter the existing centralised database, so I don't wish to run migrations for it.

So my question is this: How can I specify which to run both sets of migrations (for both connections) and when to only run one (for the local database).

I've tried `php artisan migrate --database=local_connection` but that still runs both. The centralised migrations use the `Schema::connection()` function to specify the connection. If there was some way to tell artisan to ignore any migration that made use of a specific connection, that would be great.

**Comment by x7ryan** [comment, marked as answer]
That command tells Laravel AFAIK to run **all** the migrations in `database\migrations` using the named database connection. However you mentioned that you specify a connection in the migration. So what I suspect is happening is running all migrations and the connection specified in the migration is overriding what you chose with `--database`.

What I would do is create a subfolder for the migrations for the second connection. They would then not be ran unless you specifically call the migrate command with I believe the argument is `--path=database/migrations/(sub-folder-name)` and since you are specifying the connection in the migration file I don't think you need to specify in command, therefore `php artisan migrate` should just do the local migrations directly in the migrations folder, and your test setup would need to call `php artisan migrate --path=database/migrations/(sub-folder-name)` to run the other migrations.
